### DISCUSSION TITLE
Run class from JAR with alias in catalog?

### DISCUSSION THREAD

**Discussion by axkr** [jbangdev/jbang#1364] [Q&A] [answered]
Is there a simple  jbang catalog alias example, how to run an existing class from a JAR?

**Comment by maxandersen** [comment, marked as answer]
So not the main class? 


Something like `jbang alias add --name myalgias --main other.mainClass your.jar` should work

**Reply by axkr** [reply]
I mean, how to specify (one of multiple) main classes which should be used and are not listed in the `MANIFEST.MF` of the JAR?

I tried this:
```
jbang org.matheclipse:matheclipse-io:2.0.0  --main=org.matheclipse.io.eval.MMAConsole
```

And if that works, whats best practice to define aliases for these main classes in a java `jbang-catalog.json` file?

**Reply by axkr** [reply]
Ok this approach works through a "redirect" in a Java file.
* https://github.com/axkr/jbang-catalog/commit/d672ade7deefcbad9afae9c34ba39dec5dfc92aa

Is this best practice or can the `mma` alias be configured more "directly"?

**Reply by quintesse** [reply]
@axkr with the answer @maxandersen gave you the "redirect" shouldn't be needed, right? You could run the alias and it would take care of specifying the correct main class.

**Reply by maxandersen** [reply]
> Ok this approach works through a "redirect" in a Java file.
> 
> * https://github.com/axkr/jbang-catalog/commit/d672ade7deefcbad9afae9c34ba39dec5dfc92aa
> 
> 
> 
> Is this best practice or can the `mma` alias be configured more "directly"?

Yes, using your example with my generic one:

`jbang alias add --name mmaconsole --main=org.matheclipse.io.eval.MMAConsole org.matheclipse:matheclipse-io:2.0.0`

Should give you something like this in your catalog:

```

"aliases": {
    "mmaconsole": {
      "script-ref": "org.matheclipse:matheclipse-io:2.0.0",
      "main": "org.matheclipse.io.eval.MMAConsole"
    }
  },
  ```

**Reply by axkr** [reply]
Thank you very much.

That works great:
- https://github.com/axkr/jbang-catalog/commit/f1346ae6e10fa2a7f89c1d0bec0a9f3b9bfa0f70
